docs/api-reference.md

```diff
@@ -1730,3 +1730,235 @@ Resolves a relative path to an absolute URL within the application.
+
+| Name                | Description                                                 |
+| ------------------- | ----------------------------------------------------------- |
+| [parseResponse]()   | Deserializes the response data if it contains valid JSON.   |
+| [retryOnError]()    | Retries the request if the server responds with an error.   |
+| [waitForResponse]() | Validates the response and retries if the validation fails. |
+
+##### parseResponse
+
+Deserializes the response data if it contains valid JSON. If the data is not JSON, the original string is kept. Initialized by the `createParseResponseMiddleware` factory function.
+
+This middleware takes no options. You can add it to a request as shown below:
+
+```js
+import { createParseResponseMiddleware } from 'resolve-client'
+...
+
+const { data } = await client.query(
+  {
+    name: 'articles',
+    resolver: 'all'
+  },
+  {
+    middleware: {
+      response: [createParseResponseMiddleware()]
+    }
+  }
+)
+```
+
+##### retryOnError
+
+Retries the request if the server responds with an error. Initialized by the `createRetryOnErrorMiddleware` factory function.
+
+The `retryOnError` middleware takes the following options:
+
+| Option Name | Description                                                          |
+| ----------- | -------------------------------------------------------------------- |
+| attempts    | The number of retries on error.                                      |
+| errors      | An array of error codes that are allowed to trigger a retry.         |
+| debug       | If set to `true`, the middleware logs errors to the browser console. |
+| period      | A time period to wait between retries specified in milliseconds.     |
+
+You can add the `retryOnError` middleware to a request as shown below:
+
+```js
+import { createRetryOnErrorMiddleware } from 'resolve-client'
+...
+
+client.command(
+  {
+    aggregateName: 'Chat',
+    type: 'postMessage',
+    aggregateId: userName,
+    payload: message
+  },
+  {
+    middleware: {
+      error: [
+        createRetryOnErrorMiddleware({
+          attempts: 3,
+          errors: [500],
+          debug: true,
+          period: 500
+        })
+      ]
+    }
+  },
+  err => {
+    if (err) {
+      console.warn(`Error while sending command: ${err}`)
+    }
+  }
+)
+```
+
+##### waitForResponse
+
+Validates the response and retries if the validation fails. This allows you to check whether the response contains the latest data and, if it does not, wait for a Read Model to update.
+
+Initialized by the `createWaitForResponseMiddleware` factory function.
+
+The `waitForResponse` middleware takes the following options:
+
+| Option Name | Description                                                          |
+| ----------- | -------------------------------------------------------------------- |
+| attempts    | The number of retries on validation error.                           |
```

Comment:
"retries on validation error" What do you mean when you say 'on' validation error?

Reply:
Changed to "The number of retries if validation fails.  "